diff --git a/node_modules/react-native-render-html/src/helpers/getNativePropsForTNode.ts b/node_modules/react-native-render-html/src/helpers/getNativePropsForTNode.ts
index 113b28c..415eb92 100644
--- a/node_modules/react-native-render-html/src/helpers/getNativePropsForTNode.ts
+++ b/node_modules/react-native-render-html/src/helpers/getNativePropsForTNode.ts
@@ -70,7 +70,7 @@ export default function getNativePropsForTNode<
     children,
     onPress: syntheticOnPress,
     style: [style, passedNativeProps?.style, switchProp.style],
-    testID: tnode.tagName || undefined
+    testID: switchProp.testID ||tnode.tagName || undefined
   };
   return nativeProps as any;
 }
